Removed maven flag -DarchetypeCatalog='file://../archetype-catalog.xm…

cweitkamp opened the pull request:
…l' in scripts for new bindings (see issue #2169)

Signed-off-by: Christoph Weitkamp github@christophweitkamp.de

cweitkamp commented:
Two questions:
1. Ist this the corract way to solve issue #2169?
2. If so, is the file "addons/archetype-catalog.xml" necessary anymore? Or should it be removed, too?

cweitkamp commented:
@martinvw Yes, I will do test that. Since I am not a maven expert let's clarify the next steps:

1. I clear my local maven cache (delete .m2/ folder) and run my adopted script again.
2 If it doesn't run correctly, I convert (rename the archetype-catalog.xml to archetype-settings.xml) and add the --settings flag in the create_openhab_binding_skeleton.* scripts.

Am I right? If not I need a hint where I can find the syntax for a valid settings.xml file. Thanks in advance.

martinvw commented:
To make sure you do not have to redownload the half internet you can just start by removing org/eclipse/smarthome/archetype

I think the file should look like this

```xml
<settings xmlns="http://maven.apache.org/SETTINGS/1.0.0"
      xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
      xsi:schemaLocation="http://maven.apache.org/SETTINGS/1.0.0
                          https://maven.apache.org/xsd/settings-1.0.0.xsd">
<repository>
    <id>archetype</id>
    <url>https://repository.domain.com/path/to/repo/</url>
  </repository>
</settings>
```

See also:
http://maven.apache.org/archetype/maven-archetype-plugin/generate-mojo.html
https://maven.apache.org/settings.html

cweitkamp commented:
I created a valid `archetype-settings.xml` and added the `-s` option in the `create_openhab_binding_skeleton.*` scripts. This removes the follwing warnings:
```
[WARNING] Archetype not found in any catalog. Falling back to central repository.
[WARNING] Add a repsoitory with id 'archetype' in your settings.xml if archetype's repository is elsewhere.
```
And changed to:
```
[INFO] Archetype repository not defined. Using the one from [org.eclipse.smarthome.archetype:org.eclipse.smarthome.archetype.binding:0.9.0-SNAPSHOT -> https://repo.eclipse.org/content/repositories/smarthome-snapshots] found in catalog remote
```

martinvw commented on addons/archetype-settings.xml:
```diff
@@ -0,0 +1,20 @@
+    <profiles>
+        <profile>
+            <id>smarthome-snapshots</id>
+            <repositories>
```
Is there a specific reason you added the profile, as far as I know repositories can also be used directly, you will need only half the lines in that case.

martinvw replied:
I see in my example I missed the surrounding `repositories`-tag

cweitkamp replied:
Yes there is. In the example [here](http://stackoverflow.com/questions/43513597/whats-this-bug-of-maven3-3-9-while-init-opendaylight-archetecture) and th documentation [here](https://maven.apache.org/settings.html#Repositories) the profiles are uses, too. The mvn command ignores the `repositories` or `repository` tags if not surrounded by a profile section.

martinvw replied:
A great, had that incorrect in my head :-S

martinvw commented on addons/binding/create_openhab_binding_skeleton.cmd:
```diff
@@ -11,7 +10,18 @@ IF %ARGC% NEQ 2 (
 )
 
-mvn archetype:generate -N -DarchetypeGroupId=org.eclipse.smarthome.archetype -DarchetypeArtifactId=org.eclipse.smarthome.archetype.binding -DarchetypeVersion=0.9.0-SNAPSHOT -DgroupId=org.openhab.binding -DartifactId=org.openhab.binding.%2 -Dpackage=org.openhab.binding.%2 -Dversion=2.1.0-SNAPSHOT -DbindingId=%2 -DbindingIdCamelCase=%1 -DvendorName=openHAB -Dnamespace=org.openhab
+mvn -s ../archetype-settings.xml archetype:generate -N \
```
Does this work on windows (I can test for you at home, if you like)? I always assumed there was a reason to do it like that :-)

cweitkamp replied:
Yes please. I don't have a Windows machine so your test is very much appreciated.

jwveldhuis replied:
Breaking the lines with `\` doesn't work in Windows. I just tested this script.
Use the `^` instead and it works, or put everything on one line as it was.

martinvw replied:
@jwveldhuis thanks!

cweitkamp replied:
@jwveldhuis Thanks for testing. I revert adding the linebreaks. Sry for my headlessness. Didn't thought about that.

cweitkamp commented:
@martinvw Thanks for reviews. Should this be changed in [https://github.com/eclipse/smarthome/](https://github.com/eclipse/smarthome/) too?

martinvw commented:
If the scripts exists there is wel, yes please

martinvw commented:
Yes we do: https://github.com/eclipse/smarthome/tree/master/extensions/binding
